web-services/query/src/main/java/datawave/webservice/query/logic/QueryLogic.java

```diff
@@ -23,6 +23,32 @@
 
 public interface QueryLogic<T> extends Iterable<T>, Cloneable, ParameterValidator {
     
+    /**
+     * A mechanism to get the normalized query without actually setting up the query. This can be called with having to call initialize.
+     *
+     * The default implementation is to return the query string as the normalized query
+     *
+     * @param connection
+     *            - Accumulo connector to use for this query
+     * @param settings
+     *            - query settings (query, begin date, end date, etc.)
+     * @param runtimeQueryAuthorizations
+     *            - authorizations that have been calculated for this query based on the caller and server.
+     * @param expandFields
+     *            - should unfielded terms be expanded
+     * @param expandValues
+     *            - should regex/ranges be expanded into discrete values
+     */
+    default String getPlan(Connector connection, Query settings, Set<Authorizations> runtimeQueryAuthorizations, boolean expandFields, boolean expandValues)
+                    throws Exception {
+        if (expandFields && expandValues) {
```

Comment:
||

Reply:
Actually that is what I intended, but in hindsight this should probably throw an unsupported operation exception instead as there really is no good default implementation of this.